pkg/services/baremetal/client/ssh/ssh_client.go

```diff
@@ -318,23 +318,44 @@ func (c *sshClient) GetHardwareDetailsRAM() Output {
 func (c *sshClient) GetHardwareDetailsNics() Output {
 	out := c.runSSH(`cat << 'EOF_VIA_SSH' > nic-info.sh
 #!/bin/sh
```

Comment:
I know it is strictly speaking not part of your PR, but I would update that to:

```
#!/bin/bash

trap 'echo "ERROR: A command has failed. Exiting the script. Line was ($0:$LINENO): $(sed -n "${LINENO}p" "$0")"; exit 3' ERR
set -Eeuo pipefail
```

Here is why I prefer the Bash strict mode: https://github.com/guettli/bash-strict-mode

Feel free to provide feedback to that!